docs/pages/dsCompositeIn/ds-compositeIn_L3_TestProcedure.md

```diff
@@ -0,0 +1,396 @@
+            port: 10022
+            username: "root"
+            ip: "XXX.XXX.XXX" # IP address of the device
+            password: ' '
+        - ssh_player_secondary:
+            type: "ssh"
+            port: 10022
+            username: "root"
+            ip: "XXX.XXX.XXX" # IP address of the device
+            password: ' '
+        - ssh_hal_test:
+            type: "ssh"
+            port: 10022
+            username: "root"
+            ip: "XXX.XXX.XXX" # IP address of the device
+            password: ' '
+      outbound:
+        download_url: "tftp://tftp-server.com/rack1/slot1/"    # Download location for the CPE device
+        upload_url: "sftp://server-address/home/workspace/tftp/rack1/slot1/" # Upload location
+        upload_url_base_dir: "sftp://server-address/home/workspace/tftp/rack1/slot1"
+        httpProxy:   # Local proxy if required
+        workspaceDirectory: './logs/workspace'   # Local working directory
+
+```
+
+#### Device Configuration File
+
+Example Device configuration File: `ut/host/tests/configs/deviceConfig.yml`
+
+For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_device_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_device_config.yml)
+
+Update the target directory where `HAL` binaries will be copied into the device. Also, map the profile to the source/sink settings `YAML` file path.
+
+Ensure the platform should match with the `DUT` platform in [Rack Configuration](#rack-configuration-file)
+
+```yaml
+deviceConfig:
+  cpe1:
+    platform: "stb"    # Must match the platform in example_rack_config.yml
+    model: "uk"
+    target_directory: "/tmp"  # Path where HAL binaries are copied in device
+    test:
+      profile: "../../../../profiles/sink/Sink_compositeIn.yaml"
+      player:
+        tool: "gstreamer"
+        prerequisites:
+          - export xxxx    # Pre-commands required to play the stream
+
+```
+
+#### Test Setup Configuration File
+
+Example Test Setup configuration File: `ut/host/tests/dsCompositeIn_L3_Tests/dscompositeIn_L3_testSetup.yml`
+
+Update the artifact paths from which the binaries should be copied to the device.
+
+Set the execution paths and provide the stream paths for each test case.
+
+If a test case requires multiple streams or needs to be validated using several streams, ensure that all necessary streams are added sequentially for that specific test case.
+
+```yaml
+dscompositeIn:
+  description: "dscompositeIn Device Settings test setup"
+  assets:
+    device:
+      defaults: &defaults
+        artifacts:
+          - "<path>/bin/hal_test"
+          - "<path>/bin/libut_control.so"
+          - "<path>/bin/Sink_compositeIn.yaml"
+          - "<path>/bin/run.sh"
+        execute:
+          - "chmod +x /opt/HAL/dscompositeIn_L3/hal_test"
+          - "chmod +x /opt/HAL/dscompositeIn_L3/run.sh"
+          - cp -rf /usr/lib/libdshal.so /opt/HAL/dscompositeIn_L3/
```

Comment:
should the cp command go with quotes here? ""

Reply:
done